### PR TITLE
Use GH cache for sccache on GH mac runners

### PR DESCRIPTION
**Pull request by clee2000**
sccache added GH cache as a storage option, so try to use it for the GH provided mac runners.

My experiments with this are varied.  I tried a couple of different releases and the first run with a cold cache took 1hr (v0.3.3), 1hr (v0.4.0 pre7), 2hr (v0.3.3).  

Afterwards it usually takes 30 minutes but sometimes longer, but no longer than 1hr.

I am using v0.4.0 pre7 because they reduced the amount of configuration/env vars you need to set and the GH cache keys get managed by sccache.

### CODE REVIEW

File: .github/workflows/_mac-build.yml [hunk 1]
```diff
@@ -119,10 +125,20 @@ jobs:
           max_attempts: 3
           retry_wait_seconds: 90
           command: |
-            sudo curl --retry 3 --retry-all-errors https://s3.amazonaws.com/ossci-macos/sccache_v2.15 --output /usr/local/bin/sccache
+            if [[ "${SCCACHE_USE_GHA}" == "true" ]]; then
+              sudo curl --retry 3 --retry-all-errors -OL https://github.com/mozilla/sccache/releases/download/v0.4.0-pre.7/sccache-v0.4.0-pre.7-x86_64-apple-darwin.tar.gz
```
Comment (huydhn) [inline .github/workflows/_mac-build.yml:129]: Could we cache this one on S3 `ossci-macos` bucket?  I guess it's ok because it's open source.

Comment (huydhn) [inline .github/workflows/_mac-build.yml:129]: Also `nick-fields/retry@v2.8.2` already has retries, so may be you don't need `--retry 3` here, but having it is ok too IMO.

**Review by huydhn — APPROVED** [review]
LGTM! But I think we should use S3 if possible

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**